packages/editor-ui/src/components/ResourceLocator/ResourceLocator.vue

```diff
@@ -1,115 +1,199 @@
+								/>
+								<n8n-input
+									v-else
+									:class="{[$style.selectInput]: isListMode}"
+									:size="inputSize"
+									:value="valueToDislay"
+									:disabled="isReadOnly"
+									:readonly="isListMode"
+									:title="displayTitle"
+									:placeholder="inputPlaceholder"
+									type="text"
+									ref="input"
+									@input="onInputChange"
+									@focus="onInputFocus"
+									@blur="onInputBlur"
+								>
+									<div
+										v-if="isListMode"
+										slot="suffix"
+									>
+										<i
+											:class="{
+												['el-input__icon']: true,
+												['el-icon-arrow-down']: true,
+												[$style.selectIcon]: true,
+												[$style.isReverse]: showResourceDropdown,
+											}"
+										></i>
+									</div>
+								</n8n-input>
+							</div>
+						</template>
+					</draggable-target>
+					<parameter-issues
+						v-if="parameterIssues && parameterIssues.length"
+						:issues="parameterIssues"
+					/>
+					<div v-else-if="urlValue">
+						<n8n-icon-button
+							:title="getLinkAlt(valueToDislay)"
+							icon="external-link-alt"
+							:text="true"
+							type="tertiary"
+							@click.stop="openResource(urlValue)"
+						/>
 					</div>
-				</template>
-			</DraggableTarget>
-			<parameter-issues v-if="resourceIssues" :issues="resourceIssues" />
-		</div>
-		<div :class="$style['info-text']">
-			<n8n-text
-				v-if="infoText"
-				size="small"
-			>
-					{{ infoText }}
-			</n8n-text>
-		</div>
+				</div>
+			</div>
+		</resource-locator-dropdown>
+		<parameter-input-hint v-if="infoText" class="mt-4xs" :hint="infoText" />
 	</div>
 </template>
 
 <script lang="ts">
 import mixins from 'vue-typed-mixins';
 
-import { INodeProperties, INodePropertyMode } from 'n8n-workflow';
-import { getParameterModeLabel, validateResourceLocatorParameter } from './helpers';
+import {
+	ILoadOptions,
+	INode,
+	INodeCredentials,
+	INodeListSearchItems,
+	INodeListSearchResult,
+	INodeParameterResourceLocator,
+	INodeParameters,
+	INodeProperties,
+	INodePropertyMode,
+	NodeParameterValue,
+} from 'n8n-workflow';
+import {
+	hasOnlyListMode,
+} from './helpers';
 
 import DraggableTarget from '@/components/DraggableTarget.vue';
 import ExpressionEdit from '@/components/ExpressionEdit.vue';
 import ParameterIssues from '@/components/ParameterIssues.vue';
-import { PropType } from 'vue';
+import ParameterInputHint from '@/components/ParameterInputHint.vue';
+import ResourceLocatorDropdown from './ResourceLocatorDropdown.vue';
+import Vue, { PropType } from 'vue';
+import { INodeUi, IResourceLocatorReqParams, IResourceLocatorResultExpanded } from '@/Interface';
+import { debounceHelper } from '../mixins/debounce';
+import stringify from 'fast-json-stable-stringify';
+import { workflowHelpers } from '../mixins/workflowHelpers';
+import { nodeHelpers } from '../mixins/nodeHelpers';
+import { getAppNameFromNodeName } from '../helpers';
 
+interface IResourceLocatorQuery {
```

Comment:
Can this be moved to `interfaces.ts`?

Reply:
it's only need for this local instance of caching